tests/debugging/test_debugger.py

```diff
@@ -1045,3 +1047,55 @@ def test_debugger_continue_wrapping_after_first_failure():
 
         assert not d._probe_registry[probe_nok.probe_id].installed
         assert d._probe_registry[probe_ok.probe_id].installed
+
+
+def test_debugger_redacted_identifiers():
+    import tests.submod.stuff as stuff
+
+    with debugger(upload_flush_interval=float("inf")) as d:
+        d.add_probes(
+            create_snapshot_line_probe(
+                probe_id="foo",
+                version=1,
+                source_file="tests/submod/stuff.py",
+                line=164,
+                **compile_template(
+                    "token=",
+                    {"dsl": "token", "json": {"ref": "token"}},
+                    " answer=",
+                    {"dsl": "answer", "json": {"ref": "answer"}},
+                    " pii_dict=",
+                    {"dsl": "pii_dict", "json": {"ref": "pii_dict"}},
+                    " pii_dict['jwt']=",
+                    {"dsl": "pii_dict['jwt']", "json": {"index": [{"ref": "pii_dict"}, "jwt"]}},
+                ),
+            )
+        )
+
+        stuff.sensitive_stuff("top secret")
+
+        ((msg,),) = d.uploader.wait_for_payloads()
+
+        assert (
+            msg["message"] == f"token='{REDACTED}' answer=42 "
```

Comment:
I see this 
```
REDACTED_PLACEHOLDER = r"{redacted}"
```

Where those values got uppercased?

Reply:
This is an f-string and `REDACTED` is a variable with value `{redacted}` 🙂